modules/backendscheduler/backendscheduler.go

```diff
@@ -256,7 +259,7 @@ func (s *BackendScheduler) Next(ctx context.Context, req *tempopb.NextJobRequest
 		level.Info(log.Logger).Log("msg", "assigned job to worker", "job_id", j.ID, "worker", req.WorkerId)
 
 		return resp, nil
-	default:
+	case <-timeoutCtx.Done():
```

Comment:
even if we move the timeout client side we need a line like this. the previous code did not respect the client cancelling context. i missed this in review.

Reply:
Me too.